runtime/queries/metricsview_aggregation.go

```diff
@@ -427,7 +427,7 @@ func (q *MetricsViewAggregation) buildMetricsAggregationSQL(mv *runtimev1.Metric
 			if unnestClause != "" {
 				unnestClauses = append(unnestClauses, unnestClause)
 			}
-			groupCols = append(groupCols, safeName(d.Name))
+			groupCols = append(groupCols, fmt.Sprintf("%d", len(selectCols)))
```

Comment:
Can we add a comment here and on line 440 about what this does and why it's required. I think this fix might seem confusing/redundant to future maintainers.